File: frontend/src/types/message.ts
export type MessageId = number;

export type ApiMessage = {
  messageId?: number;
  conversationId: number;
  userId: number;
  messageContent: string;
  timeStamp: string;
};

export interface Message {
  id?: number;
  conversationId: number;
  senderId: number;
  content: string;
  sentAt: Date;
}

export type SendMessage = {
  conversationId: number;
  userId: number;
  content: string;
};

File: BackendAPI/Models/Message.cs
namespace BackendAPI.Models;
using System.ComponentModel.DataAnnotations.Schema;

public class Message
{
    public int MessageId { get; set; }
    public string MessageContent { get; set; }
    public int ConversationId { get; set; } 

    [ForeignKey("ConversationId")]
    public Conversation Conversation { get; set; }
    public DateTime TimeStamp { get; set; }

    public int UserId { get; set; }
    [ForeignKey("UserId")] public User User { get; set; }

}